Marlin/src/HAL/HC32/app_config.h

```diff
@@ -0,0 +1,70 @@
+/**
+ * app_config.h is included by the hc32f460 arduino build script for every source file.
+ * it is used to configure the arduino core (and ddl) automatically according 
+ * to the settings in Configuration.h and Configuration_adv.h.
+ */
+#pragma once
+#ifndef _HC32_APP_CONFIG_H_
+#define _HC32_APP_CONFIG_H_
+
+#include "../../inc/MarlinConfigPre.h"
+
+//
+// dev mode
+//
+#if ENABLED(MARLIN_DEV_MODE)
+  #define __DEBUG 1
+  #define __CORE_DEBUG 1
+#endif
+
+//
+// Fault Handlers and Panic
+//
+
+#if ENABLED(POSTMORTEM_DEBUGGING)
+  // disable arduino core fault handler, as we define our own
+  #define CORE_DISABLE_FAULT_HANDLER 1
+#endif
+
+// force-enable panic handler so that we can use our custom one (in MinSerial)
+#define PANIC_ENABLE 1
+
+// use short filenames in ddl debug and core panic output
+#define __DEBUG_SHORT_FILENAMES 1
+#define __PANIC_SHORT_FILENAMES 1
+
+// omit panic messages in core panic output
+#define __OMIT_PANIC_MESSAGE 1
+
+//
+// Usart
+//
+
+// disable serial globals (Serial1, Serial2, ...), as we define our own
+#define DISABLE_SERIAL_GLOBALS 1
+
+// increase the size of the Usart buffers (both RX and TX)
+// NOTE: 
+// the heap usage will increase by (SERIAL_BUFFER_SIZE - 64) * "number of serial ports used"
+// if running out of heap, the system may become unstable
+//#define SERIAL_BUFFER_SIZE 256
+
+// enable support for Usart Clock Divider / Oversampling auto config
+#define USART_AUTO_CLKDIV_OS_CONFIG 1
+
+// enable USART_RX_DMA_SUPPORT core option when SERIAL_DMA is enabled
+#if ENABLED(SERIAL_DMA)
+  #define USART_RX_DMA_SUPPORT 1
+#endif
+
+//
+// Misc.
+//
+
+// redirect printf to host serial
+#define REDIRECT_PRINTF_TO_SERIAL 1
+
+// FIXME override F_CPU to PCLK1, as marlin freaks out otherwise
+#define F_CPU (SYSTEM_CLOCK_FREQUENCIES.pclk1)
```

Comment:
yes, marlin uses certain timer frequency constants `constexpr` expressions, which requires them to be constant values and why the timer rates in `timers.h` are fixed values (also with TODOs).

for this FIXME, i'm not quite sure what causes the issues, and i honestly cannot remember what exactly the issues were before i've added it, but the printer just seemed to behave weirdly. 

i think it could be that Marlin assumes the CPU frequency and timer frequencies to be equal, which isn't the case for HC32. But i really haven't looked into it too much

Reply:
I went looking back through past changes I made for STM32 and how these various clocks and variables are used.

I see that you aren't changing F_CPU in this PR, only changing where it is defined, so any clock/timer related changes should happen in another PR. Hopefully these comments are helpful, based on some of the questions I see in the comments in the HC32/timers.h file.

`F_CPU` - This has to be constexpr and is expected to be the rate of the main MCU core, as it is used throughout Marlin in macros such as DELAY_US. Setting this to PCLK1 probably means that anything using those timing macros is going to be running too quicky, unless the peripheral and core clocks are the same.

`timers.h : CYCLES_PER_MICROSECOND` - This is only used for some delays during babystepping, and should be based on the core clock, not the timer clock. I believe this should be deleted from your `timers.h`, allowing the default based on `F_CPU` in `macros.h` to be used.

`timers.h: HAL_TIMER_RATE` - This _does not_ have to be constexpr. It is only used inside the HAL to configure the timers.  I know this because a few years ago I modified the STM32 HAL to query it at runtime, since the clocks are not consistent across timers. I've linked the PR where I originally made that change below:
#19978 : Read STM32 clock rates from framework

I suspect that HAL_TIMER_RATE might have previously been used outside of the HAL, which doesn't seem to be true anymore. I'd consider that macro open to elimination if it no longer serves purpose in a HAL. The only outside usage I see is one STM32 board overriding the value...but I can't imagine that is actually correct.